### DISCUSSION TITLE
flaky path detection on bootstrapping flux

### DISCUSSION THREAD

**Discussion by wibed** [fluxcd/flux2#4367] [General] [answered]
i managed to get flux to bootstrap on a cluster, with the git source server from which it reconciliates on the cluster itself.

i used:
- hairpin route (router reflection)
- barebones git server
- following bootstrap command:
```yaml
flux bootstrap git --url=ssh://git@192.168.1.241:2222/srv/bigbird --private-key-file=/Users/wibe/.ssh/flux --path=/srv/bigbird/src/clusterA/staging/
```

i do serve the repo from the path `/srv/XXX` as seen from the command, and do experience flaky results concerning the path.
- it worked using a dynamic path, - but only works if the repo is empty
- it worked using the full path, - but creates a separate directory  at the root of the repo containing folders of the root of the repo

but it sometimes as well creates a new folder (ex.: with the full path), to place the flux-system folder in.
instead of recognizing the correct path within the repo. 
it does find it eventually but it takes some tries.

tl:dr
bootstrap ignores the path given at the end of the url.

**Comment by wibed** [comment, marked as answer]
it does not. i did not get an error for nonfunctioning configuration.
and it resulted in a timeout.

once corrected it works fine
